susePublicCloudInfoClient/lib/susepubliccloudinfoclient/infoserverrequests.py

```diff
@@ -19,30 +19,112 @@
 # <http://www.gnu.org/licenses/>.
 #
 
+import json
+import re
 import requests
+import sys
+import urllib
+import xml.etree.ElementTree as ET
+
+
+
+def __apply_filters(superset, filters):
+    # map operators to filter functions
+    filter_operations = {
+        '=': __filter_exact,
+        '~': __filter_substring,
+        '>': __filter_greater_than,
+        '<': __filter_less_than
+    }
+    # prepopulate the result set with all the items
+    result_set = superset
+    # run through the filters, allowing each to reduce the result set...
+    for a_filter in filters:
+        result_set = filter_operations[a_filter['operator']](
+            result_set,
+            a_filter['attr'],
+            a_filter['value']
+        )    
+    return result_set
+
+
+def __filter_exact(items, attr, value):
+    """select from items list where the attribute is an exact match to 'value'"""
+    # start with an empty result set
+    filtered_items = []
+    # iterate over the list of items
+    for item in items:
+        # append the current item to the result set if matching
+        if item[attr] == value:
+            filtered_items.append(item)
+    # return the filtered list
+    return filtered_items
+
+
+def __filter_substring(items, attr, value):
+    """select from items list where 'value' is a substring of the attribute"""
+    # start with an empty result set
+    filtered_items = []
+    # iterate over the list of items
+    for item in items:
+        # append the current item to the result set if matching
+        if value in item[attr]:
+            filtered_items.append(item)
+    # return the filtered list
+    return filtered_items
+
+
+def __filter_less_than(items, attr, value):
+    """select from items list where the attribute is less than 'value' as integers"""
+    # start with an empty result set
+    filtered_items = []
+    # iterate over the list of items
+    for item in items:
+        # append the current item to the result set if matching
+        if int(item[attr]) < int(value):
+            filtered_items.append(item)
+    # return the filtered list
+    return filtered_items
+
+
+def __filter_greater_than(items, attr, value):
+    """select from items list where the attribute is greater than 'value' as integers"""
+    # start with an empty result set
+    filtered_items = []
+    # iterate over the list of items
+    for item in items:
+        # append the current item to the result set if matching
+        if int(item[attr]) > int(value):
+            filtered_items.append(item)
+    # return the filtered list
+    return filtered_items
 
 
 def __form_url(
         framework,
         info_type,
-        data_format='xml',
-        region=None,
+        result_format='xml',
+        region='all',
         image_state=None,
-        server_type=None):
+        server_type=None,
+        apply_filters=None):
```

Comment:
apply_filters i snot used and since we had agreed to not do any additional filtering on the server, past what is done via the API, i.e. formation of the URL I don't think we need to pass 'apply_filters' as an argument to the function